.github/workflows/push.yaml

```diff
@@ -4,14 +4,22 @@ on: # yamllint disable-line rule:truthy
   push:
     branches: [main]
 
+# default permissions as read only
+permissions: read-all
+
 jobs:
   test-and-codecov:
+    permissions:
+      contents: read
```

Comment:
will codecov write anything? as this job will run after push, can we double check with document or anyway to test it?